build.sbt

```diff
@@ -352,7 +353,12 @@ lazy val unitJS = project
     buildInfoKeys := Seq[BuildInfoKey](
       "testsInputClassDirectory" -> (testsInput / Compile / classDirectory).value
     ),
-    mdocJS := Some(jsdocs)
+    mdocJS := Some(jsdocs),
+    MdocPlugin.mdocJSWorkerClasspath := {
+      val _ = (jsWorker / Compile / compile).value
```

Comment:
Dependency on `classDirectory` is not enough to trigger compilation.
The user's build will not include this setting (`mdocJSWorkerClasspath`) at all, it's here only because mdoc bootstraps itself